### PR TITLE
fix: listen state to receive sigterm cancellation

### PR DESCRIPTION
**Pull request by deryrahman**
move signal listener to main thread

### CODE REVIEW

File: task/bq2bq/executor/bumblebee/handler.py [hunk 1]
```diff
@@ -1,27 +1,51 @@
...
+                job_id = job.job_id
+                self.client.cancel_job(job_id)
+                logger.info(f"{job_id} successfully cancelled")
 
     def handle_job_finish(self, job) -> None:
         self._sum_slot_millis += job.slot_millis
         self._sum_total_bytes_processed += job.total_bytes_processed
+        self.is_done = True
 
     def handle_job_cancelled(self, client, job):
```
Comment (smarchone) [inline task/bq2bq/executor/bumblebee/handler.py]: this name doesnot make sense as this doesnot handle cancel. it just appends all jobs irrespective of sigterm signal.

File: task/bq2bq/executor/bumblebee/handler.py [hunk 2]
```diff
@@ -1,27 +1,51 @@
...
 
 class BigqueryJobHandler:
     def __init__(self) -> None:
         self._sum_slot_millis = 0
         self._sum_total_bytes_processed = 0
+        self.client = None
+        self.jobs = []
+        self.is_done = False
+        self.is_cancel = False
+        self._thread = threading.Thread(target=self._handle_state)
```
Comment (smarchone) [inline task/bq2bq/executor/bumblebee/handler.py]: self._thread might not be required. in the handle_sigterm callback we can just kill all jobs !

File: task/bq2bq/executor/bumblebee/handler.py [hunk 3]
```diff
@@ -8,20 +8,30 @@ class BigqueryJobHandler:
...
+            for job in self.jobs:
+                job_id = job.job_id
+                self.client.cancel_job(job_id)
+                logger.info(f"{job_id} successfully cancelled")
 
     def handle_job_finish(self, job) -> None:
         self._sum_slot_millis += job.slot_millis
         self._sum_total_bytes_processed += job.total_bytes_processed
 
     def handle_job_cancelled(self, client, job):
```
Comment (smarchone) [inline task/bq2bq/executor/bumblebee/handler.py]: rename the method as this does not handle cancellation. WDYT ?

Comment (deryrahman) [reply task/bq2bq/executor/bumblebee/handler.py]: yes, this method would do register the job, maybe naming such `register_job_to_cancel` might be suitable? any suggestion?

**Review by smarchone — APPROVED** [review]
LGTM!

**Pull request merged**